### PR TITLE
[jsk_fetch_startup] update navigation-utils.l to work auto-dock correctly.

### PR DESCRIPTION
**Pull request by knorth55**
update `auto-dock.l` to work correctly.

### CODE REVIEW

**Comment by k-okada** [comment]
LGTM, but why the fetch can autodock without this modification ? @708yamaguchi

**Comment by 708yamaguchi** [comment]
In `auto-dock.l`, \*ri\* is created when `navigation-utils.l` is loaded.
Therefore, `(require :fetch-interface "package://fetcheus/fetch-interface.l")` and `(fetch-init)` in this PR are not necessarily required.
https://github.com/jsk-ros-pkg/jsk_robot/blob/c3a53e1d8bf17585932316d1751d8bd949ba3869/jsk_fetch_robot/jsk_fetch_startup/euslisp/navigation-utils.l#L7-L10

However, I think this PR helps people to understand the code.

**Comment by knorth55** [comment]
I mean adding `:clear-costmap` at the beginning is important to start this demo smoothly.

**Comment by k-okada** [comment]
> In auto-dock.l, *ri* is created when navigation-utils.l is loaded.

ok, if so we need decide policy, whether if we should create `*ri*` within `navigation-utils.l` or not. If we decided to initialize `*ri*` outside `navigation-utils.l`, remove `(fetch-init)` within `navigation-utils.l` with comments and add `(fetch-init)` for all app programs loading `navigation-utils.l`.

**Comment by knorth55** [comment]
i meant `:clear-costmap` is the most important point in this PR.
in order to do it, we first make *ri*.

**Comment by knorth55** [comment]
I change to do `:clear-costmap` in `(auto-dock)` function in `navigation-utils.l`.

**Comment by knorth55** [comment]
I'm using this function in #1152

**Comment by knorth55** [comment]
@708yamaguchi can you review this again?

**Comment by 708yamaguchi** [comment]
@k-okada says
> ok, if so we need decide policy, whether if we should create *ri* within navigation-utils.l or not. If we decided to initialize *ri* outside navigation-utils.l, remove (fetch-init) within navigation-utils.l with comments and add (fetch-init) for all app programs loading navigation-utils.l.

I think making \*ri* inside of navigation-utils.l is no problem because we want to use many methods of \*ri* in navigation-utils.l.

Regarding `:clear-costmap`, if you think `:clear-costmap` is important,  how about using `:clear-costmap` in (go-to-spot) instead of (auto-dock)?

**Comment by 708yamaguchi** [comment]
thank you!
Looks good to me.

File: jsk_fetch_robot/jsk_fetch_startup/euslisp/navigation-utils.l [hunk 1]
```diff
@@ -9,6 +9,21 @@
 (defparameter *spots* nil)
 (defparameter *dock-spot* "/eng2/7f/room73B2-fetch-dock-front")
 (defparameter *is-charging* nil)
+(defparameter *tfl* (instance ros::transform-listener :init))
```
Comment (k-okada) [inline jsk_fetch_robot/jsk_fetch_startup/euslisp/navigation-utils.l]: please add your `*ri*` policy here....

Comment (knorth55) [reply jsk_fetch_robot/jsk_fetch_startup/euslisp/navigation-utils.l]: I have no policy about `*ri*`.
This PR does not have  any policy about `*ri*`.
What I want to do is to do `:clear-costmap` before `:move-to`.

Comment (k-okada) [reply jsk_fetch_robot/jsk_fetch_startup/euslisp/navigation-utils.l]: https://github.com/jsk-ros-pkg/jsk_robot/pull/1160#issuecomment-546595519 changed the code based on policy, please add that to source code, also update https://github.com/jsk-ros-pkg/jsk_robot/pull/1160#issue-332598336. 
we would like to why this commit makes auto dock work correctly, too.

Comment (knorth55) [reply jsk_fetch_robot/jsk_fetch_startup/euslisp/navigation-utils.l]: I mean https://github.com/jsk-ros-pkg/jsk_robot/pull/1160#issuecomment-546595519 is totally misunderstanding.
And I update this PR after the comment.

**Comment by knorth55** [comment]
Finally, this PR contains several modification.
- change the distance in `auto-dock` function
- do `clear-costmap` in every `go-to-spot` function
- pass `dock` position to improve `auto-dock` success ratio.

the comments posted in this thread are totally misunderstood about this PR, and I changed this PR after the comments.
I have no idea what the policy is, but I fix to work `auto-dock` properly.
It is nonsense to talk about coding policy in this PR.
I'm using this commit in Fetch15 now for go-to-kitchen app.

**Comment by knorth55** [comment]
@k-okada can you merge this PR?

**Pull request merged**